ingestors/kafka/src/main/scala/hydra/kafka/algebras/MetadataAlgebra.scala

```diff
@@ -72,6 +74,20 @@ object MetadataAlgebra {
     } yield algebra
   }
 
+  def test[F[_]: Sync: Concurrent]() = {
+    Ref[F].of(MetadataStorageFacade.empty).map(cache =>
+    new MetadataAlgebra[F] {
+      override def getMetadataFor(subject: Subject): F[Option[TopicMetadataContainer]] =
+        cache.get.map(_.getMetadataByTopicName(subject))
+
+      override def getAllMetadata: F[List[TopicMetadataContainer]] =
+        cache.get.map(_.getAllMetadata)
+
+      override def addMetadata(topicMetadataContainer: TopicMetadataContainer): F[Unit] =
+        cache.update(_.addMetadata(topicMetadataContainer))
+    })
+  }
+
```

Comment:
I think a better way to do this may be to create a new class called `TestMetadataAlgebra extends MetadataAlgebra` that way we can add the new method `addMetadata` to that new class rather than directly to the `MetadataAlgebra` trait and avoid having the unimplemented function below. Thoughts?

Reply:
Added TestMetadataAlgebra.